components/collection/drop/modal/DropConfirmModal.vue

```diff
@@ -106,9 +115,37 @@ watch(
   },
 )
 
-watch(sanitizedMintedNft, async (mintedNft) => {
-  if (mintedNft?.image) {
-    nftCoverLoaded.value = await preloadImage(mintedNft.image)
+watch([sanitizedMintedNft, retry], async ([mintedNft]) => {
+  if (mintedNft?.image && retry.value) {
+    try {
+      nftCoverLoaded.value = await preloadImage(mintedNft.image)
+    } catch (error) {
+      retry.value -= 1
+    }
+  }
+})
+
+watch(
+  () => props.claiming,
+  (claiming: boolean) => {
+    if (claiming) {
+      modalStep.value = 'claiming'
+    }
+  },
+)
+
+watch(moveSuccessfulDrop, (value) => {
+  if (value) {
+    modalStep.value = 'succeded'
   }
 })
+
+watch(
+  () => props.modelValue,
+  (isOpen) => {
+    if (isOpen) {
+      modalStep.value = 'email'
+    }
+  },
+)
```

Comment:
i think it would be more concise if you used a single watchEffect

Reply:
fixed